src/store/cartSlice.ts

```diff
@@ -0,0 +1,103 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+interface CartSliceState {
+  items: {
+    id: number;
+    name: string;
+    price: number;
+    totalPrice: number;
+    quantity: number;
+  }[];
+  totalAmount: number;
+  totalQuantity: number;
+}
+
+const initialState: CartSliceState = {
+  items: JSON.parse(localStorage.getItem("cart") as string) || [],
+  totalAmount: JSON.parse(localStorage.getItem("totalAmount") as string) || 0,
```

Comment:
"totalPrice"?

Reply:
name "totalPrice" is used in antoher place